server-spi/src/main/java/org/keycloak/models/RoleMapperModel.java

```diff
@@ -62,6 +62,15 @@
         return value != null ? value.stream() : Stream.empty();
     }
 
+    /**
+     * Returns {@code true}, if this object is directly assigned the given role.
+     * 
+     * @param role the role
+     * @return see description
+     * @see #hasRole(RoleModel) if you want to check whether this object is directly or indirectly assigned to a role
+     */
+    boolean hasDirectRole(RoleModel role);
```

Comment:
Would be better to have this method as default?

Reply:
@pedroigor: What do you mean with having this method as default? Having a default implementation which delegates to hasRole(Role)?

Reply:
Yes, but perhaps using `getRoleMappingsStream`.